fda/datasets/_real_datasets.py

```diff
@@ -387,3 +386,115 @@ def fetch_medflies(return_X_y: bool=False):
 
 if hasattr(fetch_medflies, "__doc__"):  # docstrings can be stripped off
     fetch_medflies.__doc__ += _medflies_descr + _param_descr
+
+_weather_descr = """
+    Daily temperature and precipitation at 35 different locations in Canada averaged 
+    over 1960 to 1994.
+
+    References:
+        Ramsay, James O., and Silverman, Bernard W. (2006), Functional Data Analysis, 
+        2nd ed. , Springer, New York.
+
+        Ramsay, James O., and Silverman, Bernard W. (2002), Applied Functional Data Analysis, 
+        Springer, New York
+"""
+
+
+def fetch_weather(return_X_y: bool = False):
+    """
+    Load the Canadian Weather dataset.
+
+    The data is obtained from the R package 'fda' from CRAN.
+
+    """
+    DESCR = _weather_descr
+
+    raw_dataset = fetch_cran(
+        "CanadianWeather", "fda",
+        version="2.4.7")
+
+    data = raw_dataset["CanadianWeather"]
+
+    weather_daily = np.asarray(data["dailyAv"])
+
+    temp_prec_daily = np.transpose(weather_daily[:, :, 0:2], axes=(1, 0, 2))
+
+    curves = FDataGrid(data_matrix=temp_prec_daily,
+                       sample_points=range(1, 366),
+                       dataset_label="Canadian Temperatures",
+                       axes_labels=["day", "temperature (ºC)", "precipitation (mm.)"])
+
+    target = data["region"]
+
+    if return_X_y:
+        return curves, target
+    else:
+        return {"data": curves,
+                "target": target,
+                "target_names": np.unique(target),
+                "target_feature_names": ["region"],
+                "DESCR": DESCR}
+
+
+if hasattr(fetch_weather, "__doc__"):  # docstrings can be stripped off
+    fetch_weather.__doc__ += _weather_descr + _param_descr
+
+_aemet_descr = """
+    Series of daily summaries of 73 spanish weather stations selected for the period 1980-2009.  The
+    dataset contains geographic information of each station and the average for the period 1980-2009
+    of daily temperature, daily precipitation and daily wind speed.
+    Meteorological State Agency of Spain (AEMET), http://www.aemet.es/. Government of Spain.
+
+    Authors:
+        Manuel Febrero Bande, Manuel Oviedo de la Fuente <manuel.oviedo@usc.es>
+
+    Source:
+        The data were obtained from the FTP of AEMET in 2009.
+"""
+
+
+def fetch_aemet(return_X_y: bool = False):
+    """
+    Load the Spanish Weather dataset.
+
+    The data is obtained from the R package 'fda.usc' from CRAN.
+
+    """
+    DESCR = _aemet_descr
+
+    raw_dataset = fetch_cran(
+        "aemet", "fda.usc",
+        version="1.3.0")
+
+    data = raw_dataset["aemet"]
+
+    fd_temp = data["temp"]
+    fd_logprec = data["logprec"]
+    fd_wind = data["wind.speed"]
+
+    regions = np.unique(data['df']['province'])
+    climate_regions = ["OCEANIC", "OCEANIC", "CONTINENTAL", "MEDITERRANEAN", "ARID", "OCEANIC", "CONTINENTAL",
```

Comment:
Really? Is there not other way? 😞

Reply:
I know...it is not a very clean way of doing it, and I am not even sure if it is even correct.
The locations of the weather stations are given in the original dataset but they do not come classified by climate, so I did it by hand. However, there are places where the climate is not very defined. For example, Cordoba is a Mediterranean or Continental climate? We can talk about it tomorrow, and if we have doubts about it, we do not include the dataset.

Reply:
Finally, I did not include target and the information about the weather stations is included in the description of the dataset.